[CO-2780] Updated users/identify users/delete document to include email as identifier (#7718)
braze-inc/braze-docs: _docs/_api/endpoints/user_data/post_user_delete.md

```
### Request parameters

| Parameter      | Required | Data Type                  | Description                                                                                      |
| -------------- | -------- | -------------------------- | ------------------------------------------------------------------------------------------------ |
| `external_ids` | Optional | Array of strings           | External identifiers for the users to delete.                                                    |
| `user_aliases` | Optional | Array of user alias object | [User aliases]({{site.baseurl}}/api/objects_filters/user_alias_object/) for the users to delete. |
| `braze_ids`    | Optional | Array of strings           | Braze user identifiers for the users to delete.                                                  |
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3  .reset-td-br-4}

### Deleting users by email

> If an `email` is specified as an identifier, an additional `prioritization` value is required in the identifier. The `prioritization` should be an array specifying which user to merge if there are multiple users found. `prioritization` is an ordered array, meaning if more than one user matches from a prioritization, then merging will not occur.

Reviewer: Should this say `which user to delete` if there are multiple users found?

> If an `email` is specified as an identifier, an additional `prioritization` value is required in the identifier. The `prioritization` should be an array specifying which user to merge if there are multiple users found. `prioritization` is an ordered array, meaning if more than one user matches from a prioritization, then merging will not occur.

Reviewer: There's also a reference to merging later on in this paragraph, not sure if that's intentional

> If an `email` is specified as an identifier, an additional `prioritization` value is required in the identifier. The `prioritization` should be an array specifying which user to delete if there are multiple users found. `prioritization` is an ordered array, meaning if more than one user matches from a prioritization, then deletion will not occur.
> 
> The allowed values for the array are: `identified`, `unidentified`, `most_recently_updated`. `most_recently_updated` refers to prioritizing the most recently updated user.

Reviewer 2: ```suggestion
If an `email` is specified as an identifier, an additional `prioritization` value is required in the identifier. The `prioritization` is an ordered array and should specify which user to delete if multiple users are found. This means deleting users will not occur if more than one user matches a prioritization.

The allowed values for the array are: `identified`, `unidentified`, `most_recently_updated`. `most_recently_updated` refers to prioritizing the most recently updated user.
```
